pkg/pillar/cmd/zedmanager/updatestatus.go

```diff
@@ -1189,50 +1189,60 @@ func doInactivate(ctx *zedmanagerContext, uuidStr string,
 	log.Infof("doInactivate for %s\n", uuidStr)
 	changed := false
 	done := false
+	uninstall := (status.PurgeInprogress != types.BRING_DOWN)
 
-	// First halt the domain
-	unpublishDomainConfig(ctx, uuidStr)
+	if uninstall || !status.IsContainer {
```

Comment:
Why doesn't this apply for a Container? Is the model:

If an AppInst. is de-activated / Deleted, Delete Domain Config and wait for Domain Status to be deleted.

This applies to both VMs and Containers.. right?

Reply:
The workflow of deactivating and deleting is different for both VM and containers.
In the case of deactivating, we are just deactivating or stopping the domain but in the case of deleting we are unpublishing the domain config.
Now in case of purge, we are following the domain delete flow for VM's which is not true for containers. This code will always hit for VM's and for containers while deleting only.